File: JewelsAndStones/JewelsAndStones.py
class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        arr = []
        for i in stones:
            if i in jewels:
                arr.append(i)
        return len(arr)


class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        count = 0 
        for i in stones:
            if i in set(jewels):
                count += 1
        return count
        

class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        return sum(map(jewels.count, stones))
        

class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        return sum(s in jewels for s in stones)

File: JewelsAndStones/JewelsAndStones.js
/**
 * @param {string} jewels
 * @param {string} stones
 * @return {number}
 */
var numJewelsInStones = function (jewels, stones) {
  let count = 0;
  for (let i of stones) {
    if (jewels.includes(i)) {
      count++;
    }
  }
  return count;
};


/**
 * @param {string} jewels
 * @param {string} stones
 * @return {number}
 */
var numJewelsInStones = function (jewels, stones) {
  let count = 0;
  for (let i = 0; i < stones.length; i++) {
    if (jewels.indexOf(stones[i]) >= 0) {
      count++;
    }
  }
  return count;
};


/**
 * @param {string} jewels
 * @param {string} stones
 * @return {number}
 */
var numJewelsInStones = function (jewels, stones) {
  return [...stones].reduce((a, e) => a + jewels.includes(e), 0);
};
